db/view/view.cc

```diff
@@ -1777,8 +1777,10 @@ future<> view_update_generator::mutate_MV(
                     mut.s->ks_name(), mut.s->cf_name(), base_token, view_token);
             local_view_update = _proxy.local().mutate_mv_locally(mut.s, *mut_ptr, tr_state, db::commitlog::force_sync::no).then_wrapped(
                     [s = mut.s, &stats, &cf_stats, tr_state, base_token, view_token, my_address, mut_ptr = std::move(mut_ptr),
-                            sem_units] (future<>&& f) {
+                            sem_units, this] (future<>&& f) mutable {
                 --stats.writes;
+                sem_units = nullptr;
+                _proxy.local().get_view_update_backlog();
```

Comment:
I was a bit shocked to realize that this function, storage_proxy::get_view_update_backlog(), not only has the name "get", not only is "const", and not only has the same name as database::get_view_update_backlog - it actually **sets** the global backlog, doesn't just get the local one. This is so confusing :-( But of course it's not your fault. Maybe if we don't rename this function you should at least add a comment before it, something like `// update the node's view backlog from this shard's updated backlog.`

It's not pretty that we use atomic variables :-( But I looked, and it seems none of these operations are locked, so maybe it's fine performance-wise. I wonder if we *really* need to call it on every single completion. Maybe we could do it only if this node's backlog changed "a lot" since last time we updated the backlog or something... But maybe we just do this right now, and maybe reconsider later.

Reply:
> I was a bit shocked to realize that this function, storage_proxy::get_view_update_backlog(), not only has the name "get", not only is "const", and not only has the same name as database::get_view_update_backlog - it actually **sets** the global backlog, doesn't just get the local one. This is so confusing :-( But of course it's not your fault. Maybe if we don't rename this function you should at least add a comment before it, something like `// update the node's view backlog from this shard's updated backlog.`

I'm planning to improve the naming problem anyway - I already have 8045bb5663f9b6ecabbf87140cbb0d4fee30153a in #18663 - I could move that commit here but maybe we can proceed with this current PR as it's quite simple and move on to #18663 

> It's not pretty that we use atomic variables :-( But I looked, and it seems none of these operations are locked, so maybe it's fine performance-wise. I wonder if we _really_ need to call it on every single completion. Maybe we could do it only if this node's backlog changed "a lot" since last time we updated the backlog or something... But maybe we just do this right now, and maybe reconsider later.

For now I'd like the backlog to be more reliable for https://github.com/scylladb/scylladb/pull/18334 which uses it. If we encounter some performance degradation, but we should for sure consider the approach you're mentioning